core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -104,6 +104,13 @@
     @Nonnull
     private Boolean caseSensitive = true;
 
+    /**
+     * Indicate whether to flatten directories in archive and just store the files. Will cause an {@link hudson.AbortException}
+     * if there are duplicate filenames selected.
+     */
+    @Nonnull
+    private Boolean flattenDirectories = false;
```

Comment:
After deserialization and before first assignment it will be null, so `Nonnull` lies.